packages/@tailwindcss-upgrade/src/template/parseConfig.ts

```diff
@@ -24,11 +34,21 @@ export async function parseConfig(
     relative = './' + relative
   }
 
-  let input = `@import 'tailwindcss';\n@config './${relative}'`
+  let userConfig = (await loadModule(fullConfigPath, __dirname, () => {}).then(
+    (result) => result.module,
+  )) as Config
```

Comment:
There is one caveat here: By not using core to read the config file to get the _old_ prefix (because if we use Core we'd throw since that prefix is no longer supported in v4), we now loose the evaluation of plugins which...might also contribute prefixes? Is that something that happens in reality?

Reply:
A plugin can contribute a prefix. In reality this probably doesn't happen. But I wouldn't be surprised if prefixes were specified in _presets_.

Reply:
Hmmm yeah we should resolve presets probably. I wish there was a nicer way to run this through core but we really want it to error like it does right now. Maybe I can use the compat APIs to resolve _only_ the config file properly

Reply:
I really don't like that we're exposing `loadMoudle` just for this case. I wonder if there's some way we can load the config without validation? Only problem I see there is it'd require adding an API to core.

Maybe this is something that we can revise once we revisit the APIs — probably just add a card for it later?

Reply:
Oh maybe we just import resolveConfig directly and use it?

Reply:
Given we can use internal API b/c it's going to get bundled into the upgrade package directly — rather than relying on exports.

Reply:
Yeah we can probably import it directly, note that right now it only uses `loadModule` from the `node` package so we have jiti added, but yeah we can add the dependencies to the upgrade package as well and then import the files we need from these other packages. I think we'll need to do this anyways now since we also want to load the presets etc.

Reply:
Okay so i changed this to use `resolveConfig` from core and to not export `loadModule` in the node package anymore and instead bundle the dependencies directly